packages/@aws-cdk/aws-synthetics/test/canary.test.ts

```diff
@@ -0,0 +1,251 @@
+    },
+    RuntimeVersion: 'syn-1.0',
+  });
+});
+
+test('Canary returns error when name is specified incorrectly', () => {
+  expect(() => new synthetics.Canary(stack, 'Canary', {canaryName: 'myCanary'})).toThrowError('Canary Name must be lowercase, numbers, hyphens, or underscores (no spaces)');
+});
+
+test('Canary name must be less than 21 characters', () => {
+  expect(() => new synthetics.Canary(stack, 'Canary', {canaryName: 'canary-name-super-long'})).toThrowError('Canary Name must be less than 21 characters');
+});
+
+test('Canary can have specified IAM role', () => {
+  // GIVEN
+  const role = new iam.Role(stack, 'role', {
+    assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
+  });
+
+  role.addToPolicy(new iam.PolicyStatement({
+    resources: ['*'],
+    actions: [
+      's3:PutObject',
+      's3:GetBucketLocation',
+      's3:ListAllMyBuckets',
+      'cloudwatch:PutMetricData',
+      'logs:CreateLogGroup',
+      'logs:CreateLogStream',
+      'logs:PutLogEvents',
+    ],
+  }));
+
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    canaryName: 'mycanary',
+    role,
+  });
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    Name: 'mycanary',
+    Code: {
+      Handler: 'index.handler',
+      Script: 'exports.handler = async () => {\nconsole.log(\'hello world\');\n};',
+    },
+    RuntimeVersion: 'syn-1.0',
+    ExecutionRoleArn: objectLike({
+      'Fn::GetAtt': arrayWith('roleC7B7E775'),
+    }),
+  });
+});
+
+test('Canary can have specified s3 Bucket', () => {
+  // GIVEN
+  const bucket = new s3.Bucket(stack, 'mytestbucket');
+
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    canaryName: 'mycanary',
+    artifactBucket: bucket,
+  });
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    Name: 'mycanary',
+    Code: {
+      Handler: 'index.handler',
+      Script: 'exports.handler = async () => {\nconsole.log(\'hello world\');\n};',
+    },
+    RuntimeVersion: 'syn-1.0',
+    ArtifactS3Location: objectLike({
+      'Fn::Join': arrayWith(arrayWith('s3://', {Ref: 'mytestbucket8DC16178'})),
+    }),
+  });
+});
+
+test('Canary can set schedule with Rate', () => {
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    canaryName: 'mycanary',
+    schedule: synthetics.Schedule.rate(Duration.minutes(3)),
+  });
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    Name: 'mycanary',
+    Code: {
+      Handler: 'index.handler',
+      Script: 'exports.handler = async () => {\nconsole.log(\'hello world\');\n};',
+    },
+    RuntimeVersion: 'syn-1.0',
+    Schedule: objectLike({ Expression: 'rate(3 minutes)'}),
+  });
+});
+
+test('Canary can set schedule with Expression', () => {
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    canaryName: 'mycanary',
+    schedule: synthetics.Schedule.expression('rate(3 minutes)'),
```

Comment:
Might be interesting to set an expression that **does not** look like one you could make with another factory...